components/Feed/index.js

```diff
@@ -70,6 +96,16 @@ class Feed extends Component {
     this.subscribe()
   }
 
+  componentWillReceiveProps (nextProps) {
+    const { query } = nextProps.url
+
+    if (query.hasOwnProperty('search') && !this.state.showSearch) {
```

Comment:
afaik this shouldn't be needed once you handle it in the constructor since you already call set state in the onSearchClick handler.